server/src/main/java/org/elasticsearch/index/mapper/ContentPath.java

```diff
@@ -58,6 +81,15 @@ public String pathAsText(String name) {
         return sb.toString();
     }
 
+    public String dottedFieldName(String name) {
```

Comment:
Could we add javadocs to explain what this method does and how it should be used? Same for the other public methods you are adding: addDottedFieldname / removeDottedFieldName